benchmark/scenarios/test/test_rate_measurer.py

```diff
@@ -20,10 +20,12 @@ def send_requests(self, rate_measurer, num_requests, num_samples):
         :param num_requests: The number of request we want to send.
         :param num_samples: The number of samples to collect.
         """
-        for i in range(num_samples * num_requests):
-            rate_measurer.request_sent()
+        call_duration = 4.567
+        for i in range(num_samples):
+            for i in range(num_requests):
+                rate_measurer.response_received(call_duration)
```

Comment:
This could use the single loop approach the other version did.
(for i in range(num_samples * num_requests))